## Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import pandas as pd

In [2]:
df= pd.DataFrame({'Study_time':[3,4.5,2.8,4,3.9,5.1,3.7],
                 'Final_score':[67,85,70,85,80,99,91]})

In [3]:
df.corr(method='pearson')

Study_time  Final_score
Study_time     1.000000     0.878005
Final_score    0.878005     1.000000

### Covariance came out to be 0.878 that means Study time and Final exam score are highly correlated and in positive direction.

## Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [4]:
df1=pd.DataFrame({'Amount_sleep':[8,7,6.8,8.2,9,4,7.7],
                 'Job_satisfaction':[8,8,5,7,6,5,9]})

In [5]:
df1.corr(method='spearman')

Amount_sleep  Job_satisfaction
Amount_sleep          1.000000          0.345512
Job_satisfaction      0.345512          1.000000

### Covariance came out to be 0.345 that means they are positively correlated but their correlation is not strong.

## Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [6]:
df2=pd.DataFrame({'Excercise_hours':[3,3.5,0,4.5,0,6,7,4],
                 'BMI':[27,27.6,28,22,15,23,18,23]})

In [7]:
df2.corr(method='spearman')

Excercise_hours       BMI
Excercise_hours         1.000000 -0.349398
BMI                    -0.349398  1.000000

In [8]:
df2.corr(method='pearson')

Excercise_hours       BMI
Excercise_hours         1.000000 -0.121534
BMI                    -0.121534  1.000000

### From both the test it can be said that Excercise hours effect the BMI of the body. With the increase in Excercise hours the BMi dewcreases that means body fat percentage decreases.

## Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [9]:
import numpy as np

In [13]:
A=list(np.random.randint(1,6,50))
B=list(np.random.randint(0,4,50))

In [14]:
df=pd.DataFrame({'TV_hours':A,'Physical_act':B})

In [16]:
df.head()

TV_hours  Physical_act
0         3             3
1         1             1
2         2             2
3         4             3
4         1             1

In [17]:
df.corr(method='pearson')

TV_hours  Physical_act
TV_hours      1.000000      0.011212
Physical_act  0.011212      1.000000

## Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

## Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [18]:
A=list(np.random.randint(10,45,30))
B=list(np.random.randint(40,500,30))

In [23]:
df1=pd.DataFrame({'Calls_per_day':A,'Calls_per_week':B})

In [24]:
df1.head()

Calls_per_day  Calls_per_week
0             16             332
1             12             439
2             32             483
3             38             442
4             18             256

In [25]:
df1.corr(method='pearson')

Calls_per_day  Calls_per_week
Calls_per_day        1.000000       -0.025028
Calls_per_week      -0.025028        1.000000